### ISSUE TITLE
there are two train function in du_MLP

### ISSUE THREAD

**Issue by csufangyu** [mdnls/icnn-numerical-experiments#1]
https://github.com/mdnls/icnn-numerical-experiments/blob/8aed88a83c873721d806e67d6e5a7d4b0b512595/icnn/nets.py#L191

https://github.com/mdnls/icnn-numerical-experiments/blob/8aed88a83c873721d806e67d6e5a7d4b0b512595/icnn/nets.py#L193

you can change the name :
`    def train(self, fit_fn, a, b, dx, tparams, verbose=True):
        '''
        Train the net to minimize a least squares objective between itself and the given fit_fn on the interval [a, b] 
        by fitting regularly sampled points [a, a+dx, a+2dx, ..., b].
        
        fit_fn - (np.ndarray) -> (np.ndarray): map an input tensor to an output tensor
        a - (torch.double): beginning of interval
        b - (torch.double): end of interval
        dx - (torch.double): step size
        tparams - TParams: parameters to use for training
        '''
        self.tparams = tparams
        
        x = torch.tensor(np.arange(a, b, dx), dtype=torch.double, device=self.device).view([-1, 1])
        y = fit_fn(x)
        self._train(x, y, tparams, verbose)
        
    def _train(self, x, y, tparams, verbose=True):
        x_pt = torch.tensor(x, dtype=torch.double, device=self.device)
        y_pt = torch.tensor(y, dtype=torch.double, device=self.device)
        
        if verbose:
            rng = trange(tparams.iters)
        else:
            rng = range(tparams.iters)
            
        if(tparams.b1 is not None):
            opt = torch.optim.Adam(params=self.parameters(), lr=tparams.lr, betas=(tparams.b1, tparams.b2))
        else: 
            opt = torch.optim.SGD(params=self.parameters(), lr=tparams.lr)
            
        losses = []
        for itr in rng:
            opt.zero_grad()
            loss_this_itr = self.loss(x_pt, y_pt)
            loss_this_itr.backward()
            losses.append(loss_this_itr.item())
            opt.step()
        return losses`
